src/ImageSharp/ImageFrame{TPixel}.cs

```diff
@@ -77,31 +84,31 @@ internal ImageFrame(Configuration configuration, int width, int height, TPixel b
             Guard.MustBeGreaterThan(height, 0, nameof(height));
             Guard.NotNull(metaData, nameof(metaData));
 
+            this.configuration = configuration;
             this.MemoryManager = configuration.MemoryManager;
             this.PixelBuffer = this.MemoryManager.Allocate2D<TPixel>(width, height, false);
-            this.Clear(configuration.ParallelOptions, backgroundColor);
-            this.MetaData = metaData;
-        }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ImageFrame{TPixel}" /> class.
-        /// </summary>
-        /// <param name="memoryManager">The <see cref="MemoryManager"/> to use for buffer allocations.</param>
-        /// <param name="size">The <see cref="Size"/> of the frame.</param>
-        /// <param name="metaData">The meta data.</param>
-        internal ImageFrame(MemoryManager memoryManager, Size size, ImageFrameMetaData metaData)
-            : this(memoryManager, size.Width, size.Height, metaData) {
+            if (!default(TPixel).Equals(backgroundColor))
```

Comment:
Here it is I think!
You need to always clear the buffer.

Reply:
I'm just looking at that exact segment. Thought it might be that! 😄